naxsi_src/naxsi_runtime.c

```diff
@@ -1096,7 +1132,37 @@ ngx_http_output_forbidden_page(ngx_http_request_ctx_t *ctx,
             json[2] = '\0';
             break;
           }
-          strcat(json, value);
+          // Escape it according to the RFC. JSON is pretty liberal: The only characters you must escape are \, ", and control codes (anything less than U+0020)
+          // Control codes should not be here
+          char* result = NULL;
+          int empty = 0;
+          if(strstr(value,"\"") || strstr(value,"\\"))
+          { 
+            //escaping JSON
+            result = replace_str(value,"\\","\\\\");
+            if(result)
+            {
+              result = replace_str(result,"\"","\\\"");
+              if(!result)
+              {
+                empty = 1;
+              }
+            }else{
+              empty = 1;  
+            } 
+
+            if(!empty && strlen(json)+strlen(result)<NGX_MAX_ERROR_STR-100-3)
```

Comment:
what is this magic number ? `NGX_MAX_ERROR_STR-100-3`

Reply:
It is this:
#define MAX_LINE_SIZE (NGX_MAX_ERROR_STR - 100)

and then minus place to terminate JSON (3 bytes, chars)

JSON termination below (3 chars):

https://github.com/nbs-system/naxsi/blob/a8dd4816460314c511f1fd3a322d882dbc2fb9db/naxsi_src/naxsi_runtime.c#L1108

I hope you follow :)

Reply:
cc @wargio

Reply:
do as i mentioned above